src/host/ut_host/ScreenBufferTests.cpp

```diff
@@ -7538,3 +7547,99 @@ void ScreenBufferTests::RectangularAreaOperations()
     VERIFY_IS_TRUE(_ValidateLinesContain(targetArea.top, targetArea.bottom, bufferChars, bufferAttr));
     VERIFY_IS_TRUE(_ValidateLinesContain(targetArea.right, targetArea.top, targetArea.bottom, bufferChar, bufferAttr));
 }
+
+void ScreenBufferTests::DelayedWrapReset()
+{
+    BEGIN_TEST_METHOD_PROPERTIES()
+        TEST_METHOD_PROPERTY(L"IsolationLevel", L"Method")
+        TEST_METHOD_PROPERTY(L"Data:op", L"{0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34}")

```

Comment:
I feel like a loop over `ops` might be easier without compromising on the tests too much. I mean, `IsolationLevel = Method` means that this won't reset the console state after every round anyways, right?

Reply:
Yeah, I know this is horrible, and I did consider using a loop, but the `IsolationLevel` does actually give us a clean state for each op, and that's necessary in these tests to prevent them interfering with each other. Otherwise we'd have to do a bunch of manual clean up which would be a bit messy.

The other advantage of this approach is that you get a separate test case for each op, so say there's a failure in three of the ops, you'll get three separate errors, and you know exactly what's broken. With a loop, the first failure would prevent the rest of the ops from running, which isn't as nice.

Reply:
I for one am a big fan of the "gross set of test properties that result in separate test runs". Pretty sure I do that all over RoundtripTests 😝

Reply:
I rather like this approach as well. I have it on my list to take the Reflow tests (which use a custom TAEF COM class to offer table-based testing where each test case gets its own name!) and make them "generic" so that other tests can rely on them instead of this, but at the end of the day it is still a way to make a hundred individual test cases in TAEF's eyes.

I have _significant beef_ with this kind of test:

```
TEST_METHOD(TestEverythingEverywhereAllAtOnce) {
    for(auto i = 0; i < 100; ++i) {
        VERIFY_FALSE(foo, "foo failed lol");
    }
}
```

which iteration failed? _good luck figuring it out._

Reply:
Yeah, some form of table-based testing would have been nice here. Ideally my `ops` array could have been declared as the property source, and each op would then trigger an individual test run. Similar to a [`Scenario Outline`](https://cucumber.io/docs/gherkin/reference/#scenario-outline) in a _Gherkin_ test, where you have a table of examples.

Reply:
Filed #14950 with a nod to this test :smile: